pkg/system/phase4_configuring.go

```diff
@@ -229,105 +232,134 @@ func (r *Reconciler) SetDesiredHPAEndpoint() {
 	}
 }
 
-// ReconcileSecretOp creates a new system in the noobaa server if not created yet.
-func (r *Reconciler) ReconcileSecretOp() error {
+// ReconcileNBSystem creates a new system in the noobaa server if not created yet.
+func (r *Reconciler) ReconcileNBSystem() error {
 	// Skip if joining another NooBaa
 	if r.JoinSecret != nil {
 		return nil
 	}
 
-	// log := r.Logger.WithName("ReconcileSecretOp")
-	util.KubeCheck(r.SecretOp)
-
-	if r.SecretOp.StringData["auth_token"] != "" {
-		r.NBClient.SetAuthToken(r.SecretOp.StringData["auth_token"])
-		return nil
+	found := util.KubeCheck(r.SecretAdmin)
+	if r.SecretAdmin.StringData["email"] == "" {
+		r.SecretAdmin.StringData["email"] = options.AdminAccountEmail
 	}
-
-	if r.SecretOp.StringData["email"] == "" {
-		r.SecretOp.StringData["email"] = options.AdminAccountEmail
+	if r.SecretAdmin.StringData["password"] == "" {
+		r.SecretAdmin.StringData["password"] = util.RandomBase64(16)
 	}
 
-	if r.SecretOp.StringData["password"] == "" {
-		r.SecretOp.StringData["password"] = util.RandomBase64(16)
-		r.Own(r.SecretOp)
-		err := r.Client.Create(r.Ctx, r.SecretOp)
-		if err != nil {
+	r.Own(r.SecretAdmin)
+	if found {
+		if err := r.Client.Update(r.Ctx, r.SecretAdmin); err != nil {
+			return err
+		}
+	} else {
+		if err := r.Client.Create(r.Ctx, r.SecretAdmin); err != nil {
 			return err
 		}
 	}
 
-	res, err := r.NBClient.CreateAuthAPI(nb.CreateAuthParams{
+	r.NooBaa.Status.Accounts.Admin.SecretRef.Name = r.SecretAdmin.Name
+	r.NooBaa.Status.Accounts.Admin.SecretRef.Namespace = r.SecretAdmin.Namespace
+
+	_, err := r.NBClient.CreateAuthAPI(nb.CreateAuthParams{
 		System:   r.Request.Name,
 		Role:     "admin",
-		Email:    r.SecretOp.StringData["email"],
-		Password: r.SecretOp.StringData["password"],
+		Email:    r.SecretAdmin.StringData["email"],
+		Password: r.SecretAdmin.StringData["password"],
 	})
-	if err == nil {
-		// TODO this recovery flow does not allow us to get OperatorToken like CreateSystem
-		r.SecretOp.StringData["auth_token"] = res.Token
-	} else {
-		res, err := r.NBClient.CreateSystemAPI(nb.CreateSystemParams{
+	if err != nil {
+		_, err := r.NBClient.CreateSystemAPI(nb.CreateSystemParams{
 			Name:     r.Request.Name,
-			Email:    r.SecretOp.StringData["email"],
-			Password: r.SecretOp.StringData["password"],
+			Email:    r.SecretAdmin.StringData["email"],
+			Password: r.SecretAdmin.StringData["password"],
 		})
 		if err != nil {
 			return err
 		}
-		// TODO use res.OperatorToken after https://github.com/noobaa/noobaa-core/issues/5635
-		r.SecretOp.StringData["auth_token"] = res.Token
 	}
-	r.NBClient.SetAuthToken(r.SecretOp.StringData["auth_token"])
-	return r.Client.Update(r.Ctx, r.SecretOp)
+
+	return nil
 }
 
-// ReconcileSecretAdmin creates the admin secret
-func (r *Reconciler) ReconcileSecretAdmin() error {
+// ReconcileSecretOp reterive and save the operator account token in a secret
+func (r *Reconciler) ReconcileSecretOp() error {
```

Comment:
Shouldn't we run this when joining a remote system?
Doesn't the remote operator need a token too?

Reply:
@guymguym Yes but this is done in another PR #207 related to remote endpoint. 
That PR is waiting to be rebased after this one is in

Reply:
OK. Resolving.